inits:
  common:
    9: [1, 0]      # return delay time
    11: [1, 3]     # operating mode
    24: [4, 10]    # moving threshold
    31: [1, 75]    # temperature limit
    32: [2, 140]   # max voltage
    44: [4, 1023]  # velocity limit
    48: [4, 4095]  # max position
    52: [4, 0]      # min position

  time-dir:
    10: [1, 0]     # velocity profile, direct

  time-inv:
    10: [1, 1]     # velocity profile, inverse rotation

buses:
  upper_body:
    port: '/dev/ttySC0'
    baud_rate: 1000000
    # processing every:
    read_pve: 1.0
    write_pve: 1.0
    read_tv: 100.0
    devices:
      head_p: 
        dev_id: 51
        inits: [common, time-inv]
      head_y:
        dev_id: 52
        inits: [common, time-dir]
      l_sho_p:
        dev_id: 34
        inits: [common, time-inv]
      l_sho_r:
        dev_id: 33
        inits: [common, time-inv]
      l_elb_y:
        dev_id: 32
        inits: [common, time-inv]
      l_elb_p:
        dev_id: 31
        inits: [common, time-inv]
      r_sho_p:
        dev_id: 44
        inits: [common, time-dir]
      r_sho_r:
        dev_id: 43
        inits: [common, time-dir]
      r_elb_y:
        dev_id: 42
        inits: [common, time-dir]
      r_elb_p:
        dev_id: 41
        inits: [common, time-inv]


  lower_body:
    port: '/dev/ttySC1'
    baud_rate: 1000000
    # processing every:
    read_pve: 1.0
    write_pve: 1.0
    read_tv: 100.0
    devices:
      l_ank_r:
        dev_id: 11
        inits: [common, time-inv]
      l_ank_p:
        dev_id: 12
        inits: [common, time-dir]
      l_kne_p:
        dev_id: 13
        inits: [common, time-inv]
      l_hip_r:
        dev_id: 14
        inits: [common, time-inv]
      l_hip_p:
        dev_id: 15
        inits: [common, time-dir]
      l_hip_y:
        dev_id: 16
        inits: [common, time-inv]
      r_ank_r:
        dev_id: 21
        inits: [common, time-inv]
      r_ank_p:
        dev_id: 22
        inits: [common, time-inv]
      r_kne_p:
        dev_id: 23
        inits: [common, time-dir]
      r_hip_r:
        dev_id: 24
        inits: [common, time-inv]
      r_hip_p:
        dev_id: 25
        inits: [common, time-inv]
      r_hip_y:
        dev_id: 26
        inits: [common, time-dir]

groups:
  head: [head_y, head_p]
  left_arm: [l_sho_p, l_sho_r, l_elb_y, l_elb_p]
  right_arm: [r_sho_p, r_sho_r, r_elb_y, r_elb_p]
  arms: [left_arm, right_arm]
  upper_body: [head, left_arm, right_arm]
  left_leg: [l_hip_y, l_hip_p, l_hip_r, l_kne_p, l_ank_p, l_ank_r]
  right_leg: [r_hip_y, r_hip_p, r_hip_r, r_kne_p, r_ank_p, r_ank_r]
  legs: [left_leg, right_leg]
  lower_body: [legs]
  all: [upper_body, lower_body]
